### PR TITLE
resolve lookup and set for importing values from json

### PR DESCRIPTION
**Pull request by robertleeplummerjr**
move toFunction into its own test, for easier debugging
add more tests, exposing issue
resolve issue which was expecting `layer` as an index, where `i` is actual index, and we need to set `this.biases[i][j]` which is a float, not overwrite it

### CODE REVIEW

File: test/base/json.js [hunk 1]
```diff
@@ -15,35 +15,74 @@ describe('JSON', () => {
...
-
-  it('toJSON()/fromJSON()', () => {
-    let output1 = net.run(input);
-    let output2 = net2.run(input);
-
-    assert.equal(JSON.stringify(output2), JSON.stringify(output1),
-      'loading json serialized network failed');
+  const input = {'0' : Math.random(), b: Math.random()};
+  describe('.toJSON()', () => {
+    describe('.layers', () => {
```
Comment (freddyC) [inline test/base/json.js:23]: Ticky-Tacky-Toey...I am not in love with the multiple describe lines, I don't mind them there but they kind of make the test output messy.  Look for comparison:

image


I don't hate it enough to ask you to change just enough to complain about it a little. (who really says `Ticky-Tacky-Toey` I mean come on)

Comment (robertleeplummerjr) [reply test/base/json.js:23]: lol
I don't prefer the output, but it is descriptive and allows us to compartmentalize the different items being exported both logically and for the human element.  I actually envisioned this expanding when more properties are potentially added here: https://github.com/BrainJS/brain.js/issues/111

**Review by freddyC — APPROVED** [review]
Looks good to me.  I complained about what bugged me but honestly it isn't a big deal, I am not going to request you change it I will just complain about it every time I get a chance until the good part of your heart feels guilty enough to change it........ or more likely I will probably forget about it.......until I have to do tests......

**Comment by robertleeplummerjr** [comment]
fair enough

File: test/base/json.js (hunk 1 above)
Comment (freddyC) [reply test/base/json.js:23]: energeticimpartialirrawaddydolphin-size_restricted
no-one-should-criticize-me

**Comment by robertleeplummerjr** [comment]
fantastic gifs

**Pull request merged**